tmtest/tmtest.go

```diff
@@ -58,11 +59,18 @@ func RunTendermint(ctx context.Context, t TestReporter, home string) (cleanup fu
 	t.Logf("Running %s pid=%d", tmpath, cmd.Process.Pid)
 
 	// Return a cleanup function, that will wait for the tendermint to stop.
-	//nolint
-	return func() {
+	// We also auto-kill when the context is Done
+	cleanup = func() {
+		fmt.Println("tendermint cleanup called")
```

Comment:
I think it's better to use `TestReporter`  (`t.Logf`) instead of directly writing to stdout.